src/drivers/X11/Fl_X11_Screen_Driver.cxx

```diff
@@ -252,6 +252,20 @@ int Fl_X11_Screen_Driver::h() {
   ;
 }
 
+// Returns the difference between the screen's actual width in pixels and the
+// scaled up width.  Used when positioning window with "east" gravity.
+
+int Fl_X11_Screen_Driver::w_dx() {
+	return fl_workarea_xywh[2] - (int)(w() * screens[0].scale);
```

Comment:
The constant index 0 (zero) in `screens[0].scale` doesn't look right. Shouldn't this be the actual screen's index?

I'd assume that we'd need the screen index as an argument in ` int Fl_X11_Screen_Driver::w_dx(int idx)` and the caller would have to set it appropriately.

Note: the same applies to `int Fl_X11_Screen_Driver::h_dy()` below.

Disclaimer: I noticed this by reviewing the code (w/o deeper knowledge of what exactly is going on here).